components/eamxx/cime_config/SystemTests/mvkxx.py

```diff
@@ -21,13 +25,195 @@
 from CIME.case.case_setup import case_setup
 from CIME.XML.machines import Machines
 
+# TODO: all the ensure_pip is a HACK
```

Comment:
Just curious, what is the right way to do this? Like, if you have a python tool that depends on a 3rd party python package, what is the right way to deliver the tool to users?

Reply:
Speaking of ensure_pip: libs like yaml or psutil may not be installed, but is it too much to assume that the user has pip installed? It's a somewhat fundamental python package, which should be shipped with the default py installation since 3.4..

Reply:
Thanks for clarifying, @mahf708 . @bartgol , I have seen many cases were pip was not provided by a python installation. Maybe they were old?